### ISSUE TITLE
Basic and Advanced Volume Options

### ISSUE THREAD

**Issue by aravindavk** [gluster/glusterd2#503]
Discussed an enhancement with @nbalacha about providing different user interface for better user experience. Use options table .tags to know if the option is advanced setting or basic.

If .tags contains "advanced", then it will be considered advanced setting. Default is "basic"(Even if not specified in .tags)

- volume set help - Shows only basic settable options
- volume set help --advanced - Shows all the options
- volume set <basic_option> <value> - Success
- volume set <advanced_option> <value> - Fail
- volume set <advanced_option> <value> --advanced  - Success (With Warning?)
- volume get - Shows all the options(basic and advanced)
- volume get --include-defaults - Shows all the options including default options

**Comment by atinmu** [comment]
@aravindavk I think we'd also need a respective github issue in glusterfs opened to introduce .tags field in the volume_options structure. Can you please file the same and tag it for Gluster 4.0?

**Comment by nbalacha** [comment]
A few comments.
The basic/advanced was @amarts idea.
I think volume get should show only the basic options as well. Otherwise it will show around 300 options at present which is something we want to avoid.

Are the default options the same as the basic options?

As per a discussion with @atinmu, the basic/advanced behaviour might be better handled by using a flag instead of a tag.

**Comment by aravindavk** [comment]
> I think volume get should show only the basic options as well. Otherwise it will show around 300 options at present which is something we want to avoid.

I meant, volume get shows all the options which are set(basic and advanced). If we show only basic option even if admin configured an advanced option doesn't make sense.

We can change this to

    volume get - Shows only basic options set by user
    volume get advanced - Shows all the options set by user

Both can have --show-defaults or --include-defaults option. This will show all the options including the ones not set by user

> As per a discussion with @atinmu, the basic/advanced behaviour might be better handled by using a flag instead of a tag.

+1

**Comment by atinmu** [comment]
@aravindavk Anything pending on the infra part of GD2 here?

**Comment by aravindavk** [comment]
I think Volume Get and CLI is pending. This patch is not merged https://github.com/gluster/glusterd2/pull/572

**Comment by atinmu** [comment]
@aravindavk Now I see #572 is closed. Can we close this issue now?

**Comment by Madhu-1** [comment]
@atinmu   #572 is Superseded by #861, once it's merged. we can close this issue.

**Comment by Madhu-1** [comment]
@atinmu  we can close this issue. #861  is merged

**Comment by prashanthpai** [comment]
@Madhu-1 I'd leave this open a filtering is pending

**Comment by Madhu-1** [comment]
@prashanthpai  sent a patch for filtering in #958

**Comment by Madhu-1** [comment]
@prashanthpai #958 is merged, can we close this issue now.

**Comment by atinmu** [comment]
Closed through #958.
